### ISSUE TITLE
`C-g` is treated specially while dict edit?

### ISSUE THREAD

**Issue by lo48576** [ueno/libskk#49]
Assume that using the keymap rule (`keymap/default.json`) below:

```json
{
    "define": {
        "keymap": {
            "C-g": null,
            "C-c": "abort",
            "C-r": "abort",
            ...
        }
    }
}
```

Type "P o y o SPC", and you will see `▼ぽよ【】` appears.
Then:

  * If I pressed `C-r`, nothing happens.
  * If I pressed `C-c`, nothing happens.
  * If I pressed `C-g`, escaped dictionary edit (`▽ぽよ` appeared).

At `▽ぽよ` state, `C-g` does nothing and both `C-c` and `C-g` work as expected (input cancelled).

Why `C-g` is treated specially?

***

Additional note in Japanese:
端的な説明: <https://mastodon.cardina1.red/@lo48576/99383492999038223>

#48 を利用して `"Escape": "abort-to-latin-passthrough"` のように設定することで vi-cooperative にしたかったのですが、そもそも辞書登録状態で abort が期待したように動いてなさそうだったため issue を立てました。
私もコードを読んでみたのですが、 `C-g` をハードコーディングで特別扱いしている箇所は見付からなかったため、何故このような挙動になっているのかわかりませんでした。
何らかの仕掛けで、辞書登録時に "abort" コマンドでなく "C-g" キーのみが影響を及ぼすようになっているのでしょうか？

**Comment by lo48576** [comment]
Rule I set by `rom-kana` seems not used during dict edit.
I think the problem is that the custom rule is not used during dict edit.

**Comment by lo48576** [comment]
https://github.com/ueno/libskk/blob/afe959be545d92137377d5e78ef366540ad3bac9/libskk/context.vala#L353-L354
https://github.com/ueno/libskk/blob/afe959be545d92137377d5e78ef366540ad3bac9/libskk/state.vala#L132-L142

When entering to dict edit mode, a new state is created and pushed, but the new state is created with `default` rule by default.
This seems to be causing the issue

I'll write pull request to fix this soon.

**Comment by ueno** [comment]
Right, thank you for investigating it.  I guess the fix itself would be trivial (adding ```state.typing_rule = typing_rule``` in ```start_dict_edit```), but it would be great if we have a test case too.
